content/components/text-input-with-tokens.mdx

```diff
@@ -0,0 +1,114 @@
+---
+title: Text input with tokens
+description: A text input with tokens is an input for a value that is a list.
+reactId: text_input_with_tokens
+figma: https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?node-id=10400%3A38612&t=sr8N32f7rXqjxzcn-4
+---
+
+<!-- TODO: come up with a better description -->
+
+import ComponentLayout from '~/src/layouts/component-layout'
+export default ComponentLayout
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224812709-4bc4d525-92e5-49e8-9a36-264af2ac1c0c.png"
+/>
+
+## Usage
+
+Use a text input with tokens for a value that is a list. Items can be added or removed from the list, but the individual items cannot be edited inline.
+
+This kind of input is most commonly used with an [autocomplete](/components/autocomplete) to let users select items from a predetermined list of options.
+
+## Anatomy
+
+Same as the [text input](/components/text-input), but the input's value is a list instead of a single string of text.
+
+## Options
+
+The text input can be configured with all of the same options as the regular [text input](/components/text-input) component.
+
+### Wrapping
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224812712-c3552b98-3064-4cac-83f0-7a83eb3e2a11.png"
+/>
+
+The list of tokens can be constrained to a single line or be wrapped to multiple lines. When wrapping is disabled and the list exceed the width of the input, the input will scroll horizontally.
+
+### Max height
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224812714-c349ff71-7cf8-400f-b2be-01ef1e8cf997.png"
+/>
+
+When tokens wrap, the height of the input can be constrained to a maximum height. Once the rows exceed the maximum height, the input will scroll vertically.
+
+### Truncation
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224812715-7382f758-3122-4da7-b742-34180876042f.png"
+/>
+
+You may choose to limit the number of tokens that are visible. However, all tokens will be visible while the text input is focused.
+
+### Token variants
+
+<img
+  width="960"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/224812717-20c412d7-24c0-4b94-a32a-7d3e7bc60468.png"
+/>
+
+You may choose any of Primer's token components, but you cannot mix different token components in a single input. You may also choose the size of the tokens and whether or not the token's remove button is shown.
+
+## Interactions
+
+### Move focus between tokens
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+  <Box flex={1}>
+    When tabbing to the input, the `<input />` element receives focus. Then, when the text cursor is at the beginning of
```

Comment:
Somehow the code comment renders an input 🤔 was this intentional?
![image](https://user-images.githubusercontent.com/18661030/225939183-3c086198-c355-4348-9a5b-e0fc62839138.png)

Reply:
Also, these videos are really hard to see. I think maybe recording just part of the story might make it more visible